src/tokio_util.rs

```diff
@@ -11,6 +11,13 @@ use tokio;
 use tokio::net::TcpStream;
 use tokio_executor;
 
+use std::sync::atomic::AtomicUsize;
+use std::sync::atomic::Ordering;
+lazy_static! {
+  // Keep unique such that no collisions in TcpListener accept task map
+  static ref NEXT_ACCEPT_ID: AtomicUsize = AtomicUsize::new(0);
```

Comment:
Can you use a HashSet instead and avoid this machinery?

Reply:
It seems that `futures::task::Task` does not implement a few traits for comparison...